### ISSUE TITLE
v3.10.0 broke the onChange for the input radio

### ISSUE THREAD

**Issue by cedmax** [developit/preact-compat#288]
The changes to fix the onChange being executed twice for the checkboxes (issue: https://github.com/developit/preact/issues/358, pr: https://github.com/developit/preact-compat/pull/247) introduced a problem with the input[type="radio"]: looking at the code it seems like you are deleting the onChange for any input type not checkbox, in favour of an onInput, but the radio wouldn't work with it.

I created a pull request for it too
https://github.com/developit/preact-compat/pull/287

**Comment by developit** [comment]
Awesome and thanks a bunch for the quick PR!

**Comment by mjsisley** [comment]
This breaks when `attribute.type === 'file'` with components such as https://github.com/odysseyscience/react-s3-uploader that use onChange.

I stopped deletion of the onChange prop with `if (attributes.type !== 'file') { delete attributes[props.onchange] };`

Firefox triggers the onChange callback twice. (Chrome and Safari work as expected, haven't tried IE/Edge).

**Comment by developit** [comment]
I think @cedmax's fix would fix that case too.

**Comment by CPatchane** [comment]
Agree with @mjsisley, the event onChange (and onInput) for `input type='file'` still doesn't fire on Chrome (v58) and Safari (v10) even using the preact-compat last version v3.11.0. But that seems to work on Firefox. Here is the code part I tested:

```
<input
    type='file'
    onChange ={e => console.log(e)}
  />
```

(That works correctly when using preact only.)

**Comment by developit** [comment]
Here's a thought: what if we changed [this line](https://github.com/developit/preact-compat/blob/master/src/index.js#L341) from its current implementation:

```js
let attr = nodeName==='input' && /^che|rad/i.test(attributes.type) ? 'onclick' : 'oninput',
```

...to:

```js
let attr = nodeName==='input' && /^fil|che|rad/i.test(attributes.type) ? 'onchange' : 'oninput',
```

... think that'd work?  IIRC `onchange` works as desired for checkbox and radio inputs like we want for `type="file"`.

**Comment by CPatchane** [comment]
Awesome, this fix seems to do the job, `onChange` events are now correctly fired on Chrome, Safari and Firefox in my test. And `onChange` events for checkbox and radio work as desired.

Thanks @developit 👍 . Looking forward to see this fix release 🎉

**Comment by developit** [comment]
Fixed on master! Waiting for a resolution to #292 before we release.
